lib/rubocop/cop/rspec_rails/timecop.rb

```diff
@@ -0,0 +1,209 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpecRails
+      # Enforces use of ActiveSupport TimeHelpers instead of Timecop.
+      #
+      # ## Migration
+      # `Timecop.freeze` should be replaced with `freeze_time` when used
+      # without arguments. Where a `duration` has been passed to `freeze`, it
+      # should be replaced with `travel`. Likewise, where a `time` has been
+      # passed to `freeze`, it should be replaced with `travel_to`.
+      #
+      # `Timecop.scale` should be replaced by explicitly calling `travel` or
+      # `travel_to` with the expected `durations` or `times`, respectively,
+      # rather than relying on allowing time to continue to flow.
+      #
+      # `Timecop.return` should be replaced with `travel_back`, when used
+      # without a block. `travel_back` accepts a block starting with Rails 6.1.
+      # For earlier Rails, where `return` is used with a block, it should
+      # be replaced by explicitly calling `freeze_time` with a block, and
+      # passing the `time` to temporarily return to.
+      #
+      # `Timecop.travel` should be replaced by `travel` or `travel_to` when
+      # passed a `duration` or `time`, respectively. As with `Timecop.scale`,
+      # rather than relying on time continuing to flow, it should be travelled
+      # to explicitly.
+      #
+      # All other usages of `Timecop` are similarly disallowed.
+      #
+      # ## RSpec Caveats
+      #
+      # Note that if using RSpec, `TimeHelpers` are not included by default,
+      # and must be manually included by updating `rails_helper` accordingly:
+      #
+      # ```ruby
+      # RSpec.configure do |config|
+      #   config.include ActiveSupport::Testing::TimeHelpers
+      # end
+      # ```
+      #
+      # Moreover, because `TimeHelpers` relies on Minitest teardown hooks,
+      # `rails_helper` must be required (instead of `spec_helper`), or a
+      # similar adapter layer must be in effect.
```

Comment:
I do not understand this part of the comments. I think what is required is that `#travel_back` is called in an `after` block in the spec configuration (be it rails_helper or spec_helper or other).

Reply:
You're right. It's not that TimeHelpers need to be required from rails_helper, or rails_helper should be required.
Minitest teardown hooks will only run for Rails example groups, those where specific metadata is set (`:model`, `:controller` etc), manually in each spec, or with `define_derived_metadata`.

This note should be changed accordingly.

Reply:
On a slightly different note, the RSpec maintainers plan to sunset setting this metadata automatically with `define_derived_metadata` sometimes in the future, I think our users will either:
1. define derived metadata on their own
2. start setting metadata explicitly

We don't know for sure, so relying on explicitly set metadata to skip or proceed with this inspection is not a viable option.

Maybe rspec-rails should wrap TimeHelpers module with their own that would inject `after { reset_time_helpers! }`? We could then write a cop that suggests using this module instead.